antiphishing/antiphishing.go

```diff
@@ -0,0 +1,183 @@
+package antiphishing
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/botlabs-gg/yagpdb/common"
+	"github.com/mediocregopher/radix/v3"
+	"github.com/sirupsen/logrus"
+)
+
+type Plugin struct {
+	stopWorkers chan *sync.WaitGroup
+}
+
+type BitFlowAntiFishResponse struct {
+	Match   bool `json:"match"`
+	Matches []struct {
+		Followed    bool    `json:"followed"`
+		Domain      string  `json:"domain"`
+		URL         string  `json:"url"`
+		Source      string  `json:"source"`
+		Type        string  `json:"type"`
+		TrustRating float64 `json:"trust_rating"`
+	} `json:"matches"`
+}
+
+var (
+	hyperphishURL      = "https://api.hyperphish.com/gimme-domains"
+	bitflowAntiFishURL = "https://anti-fish.bitflow.dev/check"
+)
+
+var logger = common.GetPluginLogger(&Plugin{})
+
+func RegisterPlugin() {
+	common.RegisterPlugin(&Plugin{
+		stopWorkers: make(chan *sync.WaitGroup),
+	})
+}
+
+func (p *Plugin) PluginInfo() *common.PluginInfo {
+	return &common.PluginInfo{
+		Name:     "AntiPhishing",
+		SysName:  "anti_phishing",
+		Category: common.PluginCategoryModeration,
+	}
+}
+
+func fetchHyperfishDomains() ([]string, error) {
+	resp, err := http.Get(hyperphishURL)
+	if err != nil {
+		return nil, err
+	}
+	domains := make([]string, 0)
+	err = json.NewDecoder(resp.Body).Decode(&domains)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	return domains, nil
+}
+
+func saveHyperfishDomains() ([]string, error) {
+	domains, err := fetchHyperfishDomains()
+	if err != nil {
+		return nil, err
+	}
+
+	//clear old domains incase there was a false positive
+	redisDelErr := common.RedisPool.Do(radix.Cmd(nil, "DEL", "hyperfish_domains"))
+	if redisDelErr != nil {
+		return nil, redisDelErr
+	}
+	//save new domains to set
+	args := append([]string{"hyperfish_domains"}, domains...)
+	redisSaveErr := common.RedisPool.Do(radix.Cmd(nil, "SADD", args...))
+	if redisSaveErr != nil {
+		return nil, redisSaveErr
+	}
+	return domains, nil
+}
+
+func queryHyperFish(input string) (*bool, error) {
```

Comment:
```suggestion
func queryHyperFish(input string) (bool, error) {
```
Typically in Go we do not return pointers to primitives when not necessary. In this case it suffices to return the zero value (`false`) upon error.